src/main/java/seedu/taassist/model/TaAssist.java

```diff
@@ -82,16 +96,44 @@ public void addStudent(Student p) {
      */
     public void setStudent(Student target, Student editedStudent) {
         requireNonNull(editedStudent);
-
         students.setStudent(target, editedStudent);
     }
 
     /**
-     * Removes {@code key} from this {@code TaAssist}.
-     * {@code key} must exist in TA-Assist.
+     * Removes {@code student} from this {@code TaAssist}.
+     * {@code student} must exist in TA-Assist.
+     */
+    public void removeStudent(Student student) {
+        requireNonNull(student);
+        students.remove(student);
+    }
+
+    //// class-level operations
+
+    /**
+     * Returns true if a class with the same identity as {@code moduleClass} exists in TA-Assist.
      */
-    public void removeStudent(Student key) {
-        students.remove(key);
+    public boolean hasModuleClass(ModuleClass moduleClass) {
+        requireNonNull(moduleClass);
+        return moduleClasses.contains(moduleClass);
+    }
+
+    /**
+     * Adds a class to TA-Assist.
+     * The class must not already exist in TA-Assist.
+     */
+    public void addModuleClass(ModuleClass moduleClass) {
+        requireNonNull(moduleClass);
+        moduleClasses.add(moduleClass);
+    }
+
+    /**
+     * Removes {@code moduleClass} from this {@code TaAssist}.
+     * {@code moduleClass} must exist in TA-Assist.
+     */
+    public void removeModuleClass(ModuleClass moduleClass) {
```

Comment:
Actually, what should the expected behaviour be if a student is assigned a module class `CS1101S`, but the `CS1101S` class is deleted? Should we unassign `CS1101S` from all students assigned to it?
This can be done in a separate issue.

Reply:
Yeap, another issue (#87) has been created for this problem!